crypto/crypto.go

```diff
@@ -27,18 +26,23 @@ func Digest(ms ...[]byte) []byte {
 	return ret
 }
 
-func GenerateKey() KeyPair {
-	pk, sk, _ := ed25519.GenerateKey(rand.Reader)
-	return KeyPair{
-		PrivateKey: sk,
-		PublicKey:  pk,
-	}
+func GenerateKey() SigningKey {
+	_, sk, _ := ed25519.GenerateKey(rand.Reader)
```

Comment:
Similar to some prev. comment, I think it makes sense that we return/handle the error here (similarly `io.ReadFull` is called on `crypto.rand` internally and in rare cases, e.g. not enough entropy available, the key generation could fail)